### DISCUSSION TITLE
Hline color problem in cross page longtblr

### DISCUSSION THREAD

**Discussion by libinxu01** [lvjr/tabularray#371] [Q&A]
I set first and last hline different color in none rowhead longtblr, but when table crosses page, the inner hlines color changes too.
I just want the first and last line with color, the inner stays default, how to achieve it ?
![1677421482922](https://user-images.githubusercontent.com/15907992/221416517-3b788fe3-bf97-421b-8648-46c42626e9d1.jpg)

```tex
\documentclass{article}
\usepackage[english]{babel}
\usepackage[letterpaper,top=2cm,bottom=2cm,left=3cm,right=3cm,marginparwidth=1.75cm]{geometry}
\usepackage{lipsum}
\usepackage{amsmath}
\usepackage{graphicx}
\usepackage[colorlinks=true, allcolors=blue]{hyperref}
\usepackage{tabularray}

\begin{document}
\section{Introduction}
\lipsum[1] \par
\lipsum[2] \par
\lipsum[3] \par
\lipsum[4] \par

\NewTblrTheme{fancy}{
  \SetTblrStyle{firsthead}{font=\bfseries}
  \SetTblrStyle{middlefoot}{\itshape}
  \SetTblrStyle{caption-tag}{red2}
}
\begin{longtblr}[
  theme = fancy,
  caption = {A Long Long Long Long Long Long Long Table},
  entry = {Short Caption},
  label = {tblr:test},
  note{a} = {It is the first footnote.},
  note{$\dag$} = {It is the second long long long long long long footnote.},
  remark{Note} = {Some general note. Some general note. Some general note.},
  remark{Source} = {Made up by myself. Made up by myself. Made up by myself.},
]{
  width = 1.0\textwidth,
  colspec = {XXX}, 
  hlines,
  hline{1}={1pt,fg=red},
  hline{Z}={1pt,fg=blue}
}
 Head    & Head  & Head    \\
 Alpha   & Beta  & Gamma   \\
 Nu      &Xi     & Omicron \\ 
 Rho     & Sigma & Tau     \\
 Phi     & Chi   & Psi     \\
 Alpha   & Beta  & Gamma   \\
 Epsilon & Zeta  & Eta     \\
 Iota    & Kappa & Lambda  \\
 Nu      &Xi     & Omicron \\ 
 Rho     & Sigma & Tau     \\
 Phi     & Chi   & Psi     \\
 Alpha   & Beta  & Gamma   \\
 Nu      &Xi     & Omicron \\ 
 Rho     & Sigma & Tau     \\
 Phi     & Chi   & Psi     \\
 Alpha   & Beta  & Gamma   \\
 Epsilon & Zeta  & Eta     \\
 Iota    & Kappa & Lambda  \\
 Nu      &Xi     & Omicron \\ 
 Rho     & Sigma & Tau     \\
 Phi     & Chi   & Psi     \\
 Alpha   & Beta  & Gamma   \\
 Nu      &Xi     & Omicron \\ 
 Rho     & Sigma & Tau     \\
 Phi     & Chi   & Psi     \\
 Alpha   & Beta  & Gamma   \\
 Epsilon & Zeta  & Eta     \\
 Iota    & Kappa & Lambda  \\
 Nu      &Xi     & Omicron \\ 
 Foot    & Foot  & Foot    \\
\end{longtblr}

\end{document}
```

**Comment by muzimuzhi** [comment]
Please make your example _complete_ (containing `\documentclass` line and `document` environment).

**Reply by libinxu01** [reply]
Updated

**Comment by muzimuzhi** [comment]
Seems it's a limitation of current `tabularray`: the counting of hline(s) are reset per page. A simplified example
```tex
\documentclass{article}
\usepackage[paperheight=5cm, noheadfoot]{geometry}
\usepackage{tabularray}
\usepackage{xcolor}

\begin{document}
\begin{longtblr}[
  caption = {A Long Table},
]{
  width = 1.0\textwidth,
  colspec = {XXX}, 
  hlines,
  hline{1}={1pt,fg=red},
  hline{Z}={1pt,fg=blue}
}
 Head    & Head  & Head    \\
 Alpha   & Beta  & Gamma   \\
 Nu      &Xi     & Omicron \\ \pagebreak 
 Rho     & Sigma & Tau     \\
 Phi     & Chi   & Psi     \\
\end{longtblr}
\end{document}
```
![image](https://user-images.githubusercontent.com/6376638/221421407-df278cfb-e6cb-484d-a592-a8153c66b052.png)

**Reply by muzimuzhi** [reply]
Now it's more like a bug because (in a five-row `longtblr`) setting `hline{6}={1pt,fg=blue}` colors last hlines on each page in blue.

**Reply by muzimuzhi** [reply]
Ok it's caused by how `\l__tblr_row_head_box` and `\l__tblr_row_foot_box` are constructed (by [`\__tblr_build_row_head_foot:`](https://github.com/lvjr/tabularray/blob/a86c8fa4a4da95345e305810c352cc757776e366/tabularray.sty#L5322-L5347)), using corresponding values of options `rowhead` and `rowfoot`.  By default `rowhead` is 0, thus the `\l__tblr_row_head_box` inserted at beginning of every sub-table, contains the first hline. Similarly with `rowfoot` set 0, `\l__tblr_row_foot_box` contains the last hline and is used at the end of each sub-table.

Note currently, for every `longtblr`, these two boxes, once constructed, are constant. Ideally, we need `first`, `middle`, and `last` variants for each of them.

`tabularray` does provide templates `firsthead`, `middlehead`, and `lasthead`, but they're used to typeset caption material, not rows and (hv)lines.

Update: In that case, `longtblr` is not as flexible as `longtable` yet.  `longtable` provides `\endhead`, `\endfirsthead`, `\endfoot`, and `\endlastfoot` to specify different head and foot for different sub-tables. In OP's example, the `hline{1}={1pt,fg=red}` is expected to only used by first head and `hline{X}={1pt,fg=blue}` by last foot.

Perhaps it's already tracked by some issue...

**Reply by lvjr** [reply]
You may see issue #130 for the reason of current design, and discussion #50 for a workaround.

**Reply by libinxu01** [reply]
> You may see issue #130 for the reason of current design, and discussion #50 for a workaround.

thanks a lot
